authorization-common/src/main/java/com/sequenceiq/authorization/service/ResourceCrnPermissionChecker.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.authorization.service;
+
+import java.lang.annotation.Annotation;
+
+import javax.inject.Inject;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByResourceCrn;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.cloudbreak.auth.security.internal.ResourceCrn;
+
+@Component
+public class ResourceCrnPermissionChecker implements PermissionChecker<CheckPermissionByResourceCrn> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceCrnPermissionChecker.class);
+
+    @Inject
+    private CommonPermissionCheckingUtils commonPermissionCheckingUtils;
+
+    @Override
+    public <T extends Annotation> Object checkPermissions(T rawMethodAnnotation, AuthorizationResourceType resourceType, String userCrn,
+            ProceedingJoinPoint proceedingJoinPoint, MethodSignature methodSignature, long startTime) {
+        CheckPermissionByResourceCrn methodAnnotation = (CheckPermissionByResourceCrn) rawMethodAnnotation;
+        String resourceCrn = commonPermissionCheckingUtils.getParameter(proceedingJoinPoint, methodSignature, ResourceCrn.class, String.class);
```

Comment:
There is another permissionchecker for resourcecrnlist if it is a collection, if someone annotates two string parameters with resourcecrn, i think the first one will be checked only, i can make an exception if someone provided more crns, because everyone should use a collection in this case i think